src/libraries/System.Net.Http/tests/FunctionalTests/SocketsHttpHandlerTest.Cancellation.cs

```diff
@@ -1,59 +1,88 @@
 // Licensed to the .NET Foundation under one or more agreements.
 // The .NET Foundation licenses this file to you under the MIT license.
 
-using System;
-using System.Collections.Generic;
 using System.Diagnostics;
 using System.IO;
-using System.Linq;
 using System.Net.Test.Common;
-using System.Text;
 using System.Threading;
 using System.Threading.Tasks;
 using Xunit;
 using Xunit.Abstractions;
 
 namespace System.Net.Http.Functional.Tests
 {
-    public abstract class HttpClientHandler_Http11_Cancellation_Test : HttpClientHandler_Cancellation_Test
+    public abstract class SocketsHttpHandler_Cancellation_Test : HttpClientHandler_Cancellation_Test
     {
-        protected HttpClientHandler_Http11_Cancellation_Test(ITestOutputHelper output) : base(output) { }
+        protected SocketsHttpHandler_Cancellation_Test(ITestOutputHelper output) : base(output) { }
+
+        private async Task ValidateConnectTimeout(HttpMessageInvoker invoker, Uri uri)
+        {
+            var sw = Stopwatch.StartNew();
+            await Assert.ThrowsAnyAsync<OperationCanceledException>(() =>
+                invoker.SendAsync(TestAsync, new HttpRequestMessage(HttpMethod.Get, uri) { Version = UseVersion }, default));
+            sw.Stop();
+
+            Assert.InRange(sw.ElapsedMilliseconds, 500, 85_000);
```

Comment:
Nit: this seems to be baking in an assumption about how ConnectTimeout is set.  If you want to hardcode it, consider making them optional parameters with these as the default values, or something like that.